moonraker/components/td1.py

```
@@ -0,0 +1,337 @@
+                    except asyncio.CancelledError:
+                        raise asyncio.CancelledError
+
+            # Detect disconnected devices
+            disconnected_serials = self._known_serials - current_serials
+            self.logger.debug(disconnected_serials, self._known_serials)
+            for serial_number in disconnected_serials:
+                if serial_number in self.serial_tasks:
+                    await self.close_device(self.serial_tasks[serial_number])
+                    del self.serial_tasks[serial_number]
+                self._known_serials.remove(serial_number)
+
+    def _find_td1_devices(self) -> list:
+        found = []
+        for port in serial.tools.list_ports.comports():
+            if port.vid == self._vid and port.pid == self._pid:
+                if port.serial_number:
+                    found.append((port.device, port.serial_number))
+        return found
+
+    async def _start_serial_task(self, port: str, serial_number: str) -> None:
+        try:
+            self.serial_tasks[serial_number] = TD1Protocol(self._config,
+                                                           serial_number,
+                                                           port,
+                                                           self.logger)
+            ret = self.serial_tasks[serial_number].initialize()
+            if ret is not None:
+                await ret
+
+            if not self.serial_tasks[serial_number].error_state:
+                self.logger.info(f"TD1 listening on {port} (Serial: {serial_number})")
+                self._known_serials.add(serial_number)
+            else:
+                self.logger.error(
+                    f"Failed to open port:{port} for TD1 serial {serial_number}")
+
+        except serial.serialutil.SerialException as e:
+            self.logger.error(f"Failed to open serial port {port}: {e}")
+            raise serial.serialutil.SerialException
```

This should be `raise` or `raise e` if you want to re-raise an exception.  It isn't valid to raise the type.